src/platform/DefaultPlatformFunctions.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Interface encapsulating various utility functions whose
+ * implementations depend on the platform / JS runtime.
+ */
+class DefaultPlatformFunctions {
+  _UNAME_CACHE: Promise<string> | null;
+
+  constructor() {
+    this._UNAME_CACHE = null as Promise<string> | null;
+  }
+
+  /**
+   * Calls Node's built-in `exec` function.
+   * If the `child_process` module is not available, return an error.
+   */
+  safeExec(
```

Comment:
do we need this public/separate?

Reply:
Moved this functionality to `getUname` since that's the only callsite